src/components/MapboxMap/MapboxMap.vue

```diff
@@ -107,54 +112,134 @@ function initMap(accessToken: string) {
         [] as string[]
       );
+
+      map.addSource("clustered-points", {
+        type: "geojson",
+        data: mapStore.createManualClusters(mapStore.geoJsonSpaces),
+        promoteId: "spaceSlug",
+      });
+
+      map.addLayer({
+        id: CLUSTERS_LAYER_ID,
+        type: 'symbol',
+        source: 'clustered-points',
+        filter: ['all', ['has', 'point_count']],
+        layout: {
+          'icon-image': ['concat', 'pill-', ['get', 'buildingOccupancy']],
+          'icon-allow-overlap': true,
+          'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
+          'text-field': [
+            'format',
+            [
+              'case',
+              ['<', ['get', 'point_count_abbreviated'], 10],
+              ['concat', '0', ['to-string', ['get', 'point_count_abbreviated']]],
+              ['to-string', ['get', 'point_count_abbreviated']],
+            ],
+            {
+              'text-color': '#FFF',
+            },
+            '            ',
+            {},
+            ['get', 'buildingSlugModified'],
+            {
+              'text-color': '#000',
+            },
+          ],
+          'text-size': 15,
+          'text-anchor': 'left',
+          'text-justify': 'left',
+          'text-transform': 'uppercase',
+          'text-offset': [-4.25, 0],
+          'text-allow-overlap': true,
+          'text-optional': true,
+        },
+      });
+
+
+
       map.addLayer({
-        id: "points",
+        id: UNCLUSTERED_LAYER_ID,
         interactive: true,
         type: "symbol",
-        source: {
-          type: "geojson",
-          data: mapStore.geoJsonSpaces,
-          promoteId: "spaceSlug",
-        },
+        source: "clustered-points",
+        filter: ["all", ["!", ["has", "point_count"]], [">=", ["zoom"], 17]],
         layout: {
           "icon-image": [
-            "match", // A rule to determine the icon for the point...
+            "match",// A rule to determine the icon for the point...
             ["get", "buildingOccupancy"], // ... is that if the `buildingOccupancy` property matches...
             ...occupancyIconNamePairs, // ... the first element of the pair from this (flat) sequence, the second element defines the name of the icon
             "map-marker-unknown", //And the final element determines the default icon
           ],
           "icon-allow-overlap": true,
         },
       });
-
+      
       restoreMapState();
       fixInsecureLinks();
       mapStore.setMap(map);
     });
   });
-  map.on("click", (e: any) => {
+
+  map.on("click", CLUSTERS_LAYER_ID, (e) => {
+    const features = map.queryRenderedFeatures(e.point, {
+      layers: [CLUSTERS_LAYER_ID],
+    });
+
+    if (features.length) {
+      const geometry = features[0].geometry;
+
+      if (geometry.type === "Point" && geometry.coordinates) {
+        const coordinates = geometry.coordinates.slice() as [number, number];
```

Comment:
Is this needed as a guard against changing the original `geometry.coordinates` inside the `map.flyTo`? Maybe then consider a more semantic
```
const [lng, lat] = geometry.coordinates; 
map.flyTo({
     center: {lng, lat}, //OR center: [lng, lat], mapbox is OK with both an object and a tuple
...
```

Reply:
This guard was needed because coordinates is not always present at a geometry typescript told me :o